model_api/cpp/models/src/classification_model.cpp

```diff
@@ -184,14 +255,58 @@ std::unique_ptr<ResultBase> ClassificationModel::get_multilabel_predictions(Infe
     result->topLabels.reserve(labels.size());
     for (size_t i = 0; i < labels.size(); ++i) {
         float score = sigmoid(logitsPtr[i]);
-        if (score > 0.5f) {
+        if (score > confidence_threshold) {
             result->topLabels.emplace_back(i, labels[i], score);
         }
     }
 
     return retVal;
 }
 
+std::unique_ptr<ResultBase> ClassificationModel::get_hierarchical_predictions(InferenceResult& infResult) {
+    const ov::Tensor& logitsTensor = infResult.outputsData.find(outputNames[0])->second;
+    const float* logitsPtr = logitsTensor.data<float>();
+
+    std::vector<std::string> predicted_labels;
+    std::vector<float> predicted_scores;
+    std::vector<float> activated_logits(hierarchical_config.num_single_label_classes);
+    std::copy(logitsPtr, logitsPtr + hierarchical_config.num_single_label_classes, activated_logits.data());
```

Comment:
```suggestion
    std::vector<float> activated_logits(logitsPtr, logitsPtr + hierarchical_config.num_single_label_classes);
```
You only use `activated_logits`'s pointer to data later. Why not use `logitsPtr` directly? Just remove `const` form `const float* logitsPtr = logitsTensor.data<float>();`

Reply:
I didn't pay attention to this at the moment when I approved. What about
> You only use `activated_logits`'s pointer to data later. Why not use `logitsPtr` directly? Just remove `const` form `const float* logitsPtr = logitsTensor.data<float>();`

Reply:
I thought you mean avoiding direct copy call and use constructor of activated_logits for that